hermes/src/store.rs

```diff
@@ -288,14 +295,365 @@ impl Store {
+            hashers::keccak256_160::Keccak160,
+            messages::{
+                Message,
+                PriceFeedMessage,
+            },
+            wire::v1::{
+                AccumulatorUpdateData,
+                Proof,
+                WormholeMerkleRoot,
+            },
+        },
+        rand::seq::SliceRandom,
+        serde_wormhole::RawMessage,
+        tokio::sync::mpsc::Receiver,
+    };
+
+    /// Generate list of updates for the given list of messages at a given slot with given sequence
+    ///
+    /// Sequence in Vaas is used to filter duplicate messages (as by wormhole design there is only
+    /// one message per sequence)
+    pub fn generate_update(messages: Vec<Message>, slot: Slot, sequence: u64) -> Vec<Update> {
+        let mut updates = Vec::new();
+
+        // Accumulator messages
+        let accumulator_messages = AccumulatorMessages {
+            slot,
+            raw_messages: messages
+                .iter()
+                .map(|message| pythnet_sdk::wire::to_vec::<_, byteorder::BE>(message).unwrap())
+                .collect(),
+            magic: [0; 4],
+            ring_size: 100,
+        };
+        updates.push(Update::AccumulatorMessages(accumulator_messages.clone()));
+
+        // Wormhole merkle update
+        let merkle_tree = MerkleTree::<Keccak160>::from_set(
+            accumulator_messages.raw_messages.iter().map(|m| m.as_ref()),
+        )
+        .unwrap();
+
+        let wormhole_message = WormholeMessage::new(WormholePayload::Merkle(WormholeMerkleRoot {
+            slot,
+            ring_size: 100,
+            root: merkle_tree.root.as_bytes().try_into().unwrap(),
+        }));
+
+        let wormhole_message =
+            pythnet_sdk::wire::to_vec::<_, byteorder::BE>(&wormhole_message).unwrap();
+
+        let vaa = Vaa {
+            nonce: 0,
+            version: 0,
+            sequence,
+            timestamp: 0,
+            signatures: vec![],    // We are bypassing signature check now
+            guardian_set_index: 0, // We are bypassing signature check now
+            emitter_chain: Chain::Pythnet,
+            emitter_address: Address(pythnet_sdk::ACCUMULATOR_EMITTER_ADDRESS),
+            consistency_level: 0,
+            payload: serde_wormhole::RawMessage::new(wormhole_message.as_ref()),
+        };
+
+        updates.push(Update::Vaa(serde_wormhole::to_vec(&vaa).unwrap()));
+
+        updates
+    }
+
+    /// Create a dummy price feed base on the given seed for all the fields except
+    /// `publish_time` and `prev_publish_time`. Those are set to the given value.
+    pub fn create_dummy_price_feed_message(seed: u8, publish_time: i64) -> PriceFeedMessage {
+        PriceFeedMessage {
+            feed_id: [seed; 32],
+            price: seed as _,
+            conf: seed as _,
+            exponent: 0,
+            ema_conf: seed as _,
+            ema_price: seed as _,
+            publish_time,
+            prev_publish_time: publish_time,
```

Comment:
fwiw you will probably need to set this field to a different value from publish_time in the future, because I think the "retrieve VAA for time" function needs to use this to disambiguate between the multiple updates at the same publish_time

Reply:
Let's make this change in the future that we add the benchmarks endpoint. I'll add a comment explaining it.

Reply:
👍